### PR TITLE
Remove permissive network policies and enable deny by default

### PR DESCRIPTION
**Pull request by mcharriere**
Signed-off-by: Matias Charriere <matias@giantswarm.io>

Towards https://github.com/giantswarm/roadmap/issues/3415

### Please check if PR meets these requirements

- [ ] Results of the diffs have been examined and no unintended changes are being introduced.

### Helper

* to disable the GH Action generating manifests diffs for installations, between target and source branches, comment the `/no_diffs_printing` on the PR.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: helm/cluster-azure/templates/helmreleases/network-policies.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
```
Comment (AndiDog) [inline helm/cluster-azure/templates/helmreleases/network-policies.yaml:1]: This is already part of the cluster chart ([see here](https://github.com/giantswarm/cluster/blob/main/helm/cluster/templates/apps/network-policies.yaml)), so we must not add this to provider-specific charts one by one.

Let's make this consistent with `cluster-aws` (except that you may not need provider-specific settings?? [example in cluster-aws which has some extra stuff for proxy-private clusters](https://github.com/giantswarm/cluster-aws/blob/67995d3415cad36a6d0e0d55e70a51037622ba81/helm/cluster-aws/templates/_network-policies_helmrelease_config.yaml)).

**Review by AndiDog — CHANGES_REQUESTED** [review]

File: helm/cluster-azure/templates/helmreleases/network-policies.yaml (hunk 1 above)
Comment (mcharriere) [reply helm/cluster-azure/templates/helmreleases/network-policies.yaml:1]: Yes thank you. I'm aware. This PR should be a draft; sorry for the noise.

**Comment by mcharriere** [comment]
This requires https://github.com/giantswarm/cluster-apps-operator/pull/492

**Review by kopiczko — APPROVED** [review]

**Comment by mcharriere** [comment]
@AndiDog could you please take a look at this PR again?

**Comment by kopiczko** [comment]
```
  {"level":"info","ts":"2024-05-15T08:44:45Z","msg":"App status for 't-e7duny04ccfpppbfdu-kube-prometheus-stack' is not yet as expected: expectedStatus='deployed' actualStatus='pending-install' (reason: 'Initial install underway')"}
  {"level":"info","ts":"2024-05-15T08:44:45Z","msg":"App status for 't-e7duny04ccfpppbfdu-prometheus-agent' is as expected: expectedStatus='deployed' actualStatus='deployed'"}
  {"level":"info","ts":"2024-05-15T08:44:45Z","msg":"App status for 't-e7duny04ccfpppbfdu-prometheus-operator-crd' is as expected: expectedStatus='deployed' actualStatus='deployed'"}
  {"level":"info","ts":"2024-05-15T08:44:45Z","msg":"App status for 't-e7duny04ccfpppbfdu-promtail' is as expected: expectedStatus='deployed' actualStatus='deployed'"}
  [FAILED] in [It] - /app/internal/common/apps.go:89 @ 05/15/24 08:44:46.724
• [FAILED] [1500.234 seconds]
Common tests observability-bundle apps [It] all observability-bundle apps are deployed without issues
/app/internal/common/apps.go:66

  [FAILED] Timed out after 1500.001s.
  Expected
      <bool>: false
  to be true
  In [It] at: /app/internal/common/apps.go:89 @ 05/15/24 08:44:46.724
```

**Comment by erkanerol** [comment]
There are some flakiness in e2e tests and I am on them now. FYI.

**Comment by erkanerol** [comment]
It seems the latest fixes work :D

**Pull request merged**